docs/features/incremental-generators.md

```diff
@@ -0,0 +1,485 @@
+# Incremental Generators
+
+## Summary
+
+Incremental generators are intended to be a new API that exists alongside
+[source generators](generators.md) to allow users to implement high performance
+incremental generation strategies.
```

Comment:
Is it about making the generators themselves high performance or allowing the hosting layer to be performant?